### PR TITLE
[Upgrade Customer Scenario] Pytest conversion for syncplan test cases

### PR DESCRIPTION
**Pull request by devendra104**
The purpose of this PR to convert the syncplan unit test(Customer upgrade scenario) to pytest and check the update of all the sync interval types.

Test Results:

**Pre-Upgrade**

```
$ pytest -s -m "pre_upgrade" tests/upgrades/test_syncplan.py
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
plugins: rerunfailures-9.1.1, forked-1.1.3, mock-1.10.4, xdist-1.34.0, services-1.3.1, cov-2.8.1
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_syncplan.py .
================================================================================================ warnings summary =================================================================================================
============================================================================= 1 passed, 1 deselected, 7 warnings in 102.07s (0:01:42) =============================================================================
```

**Post-Upgrade**

```
$ pytest -s -m "post_upgrade" tests/upgrades/test_syncplan.py 
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.6.6, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
plugins: rerunfailures-9.1.1, forked-1.1.3, mock-1.10.4, xdist-1.34.0, services-1.3.1, cov-2.8.1
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     
.
================================================================================================ warnings summary =================================================================================================
================================================================================== 1 passed, 1 deselected, 7 warnings in 44.76s ===================================================================================

```

### CODE REVIEW

**Review by mshriver — APPROVED** [review]

File: tests/upgrades/test_syncplan.py [hunk 1]
```diff
@@ -14,100 +14,88 @@
...
-        self.assertIsNone(product.sync_plan)
-        with self.assertRaises(HTTPError):
-            sync_plan.read()
+        assert sync_plan.interval == SYNC_INTERVAL[sync_interval]
+    sync_plan.delete()
+    product = product.read()
+    assert product.sync_plan is None
+    with pytest.raises(HTTPError):
+        sync_plan.read()
+    product.delete()
```
Comment (mshriver) [inline tests/upgrades/test_syncplan.py:100]: If these deletes aren't really part of the test coverage, consider moving them into the `teardown` phase by wrapping their call with `request.addfinalizer(product.delete)` and move the call up in the test function to right after the `product` is created.

Comment (devendra104) [reply tests/upgrades/test_syncplan.py]: Fixed

**Review by mshriver — APPROVED** [review]

**Comment by devendra104** [comment]
Test Results:

**Pre-Upgrade**
```
$ pytest -s -m "pre_upgrade" tests/upgrades/test_syncplan.py

=============================================================================================== test session starts ===============================================================================================
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_syncplan.py .

================================================================================================ warnings summary =================================================================================================
================================================================================== 1 passed, 1 deselected, 5 warnings in 30.45s ===================================================================================

```
**Post-Upgrade**
```
$ pytest -s -m "post_upgrade" tests/upgrades/test_syncplan.py
=============================================================================================== test session starts ===============================================================================================
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                     

tests/upgrades/test_syncplan.py .

================================================================================================ warnings summary =================================================================================================
================================================================================== 1 passed, 1 deselected, 5 warnings in 26.83s ===================================================================================
```

File: tests/upgrades/test_syncplan.py [hunk 2]
```diff
@@ -15,33 +15,22 @@
...
 from robottelo.constants import SYNC_INTERVAL
 from robottelo.datafactory import valid_cron_expressions
-from robottelo.test import APITestCase
 
 
-class ScenarioSyncPlan(APITestCase):
-    """The test class contains pre-upgrade and post-upgrade scenarios to test
+class TestSyncPlan:
+    """
+    The test class contains pre-upgrade and post-upgrade scenarios to test
```
Comment (jyejare) [inline tests/upgrades/test_syncplan.py:28]:
```suggestion
    The test class contains pre-upgrade and post-upgrade steps to test
```

File: tests/upgrades/test_syncplan.py [hunk 3]
```diff
@@ -54,60 +43,54 @@ def test_pre_sync_plan_migration(self):
...
-        }
-        create_dict(scenario_dict)
+        assert product.sync_plan.id == sync_plan.id
 
     @post_upgrade(depend_on=test_pre_sync_plan_migration)
-    def test_post_sync_plan_migration(self):
-        """Post-upgrade scenario that tests existing sync plans are working as
-        expected after satellite upgrade with migrating from pulp to katello
+    def test_post_sync_plan_migration(self, request, dependent_scenario_name):
+        """After upgrade, Sync interval update should work on existing sync plan(created before
```
Comment (jyejare) [inline tests/upgrades/test_syncplan.py:59]:
```suggestion
        """After upgrade, Sync interval of existing(created before upgrade) sync plan can be updated.
```

File: tests/upgrades/test_syncplan.py [hunk 4]
```diff
@@ -54,60 +43,54 @@ def test_pre_sync_plan_migration(self):
...
+        upgrade)
 
         :id: badaeec2-d42f-41d5-bd85-4b23d6d5a724
 
         :steps:
             1. Verify sync plan exists and works as earlier
+            2. Check the all available sync_interval type update with pre-created sync_plan
```
Comment (jyejare) [inline tests/upgrades/test_syncplan.py:66]: The check steps are really needed in steps. You can move them to expected results as example:
```
1. All available sync_intervals type should be updated in pre-created sync_plan.
```

**Review by jyejare — APPROVED** [review]
ACK with minor comments.

File: tests/upgrades/test_syncplan.py (hunk 2 above)
Comment (devendra104) [reply tests/upgrades/test_syncplan.py:28]: I think "scenario" is more appropriate than steps.

File: tests/upgrades/test_syncplan.py (hunk 3 above)
Comment (devendra104) [reply tests/upgrades/test_syncplan.py:59]: I think the statement is correct.

File: tests/upgrades/test_syncplan.py (hunk 2 above)
Comment (jyejare) [reply tests/upgrades/test_syncplan.py:28]: When we combine Pre-Steps and Post_steps, it becomes `a Scenario`. PreUpgrade and PostUpgrade are not `scenarios` but steps of one test scenario :)

**Comment by devendra104** [comment]
@jyejare Please merge this PR.

File: tests/upgrades/test_syncplan.py (hunk 2 above)
Comment (devendra104) [reply tests/upgrades/test_syncplan.py]: Pre_upgrade + Post_upgrade --> Scenario not steps

Comment (jyejare) [reply tests/upgrades/test_syncplan.py]: @devendra104 Ohh its that way then just do s/scenarios/scenario/.

**Pull request merged**